File: MLR-spring/src/main/java/com/mlr/entity/LeakageAlert.java
package com.mlr.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "leakage_alerts")
@Getter @Setter @NoArgsConstructor
public class LeakageAlert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id")
    private String userId;

    private String ruleName;

    private String message;

    private int severity; // 0=Low, 1=Medium, 2=High

    private Double amount;

    private boolean isRead = false;

    private LocalDateTime createdAt = LocalDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
}

File: MLR/DAO/Entities/LeakageAlert.cs
using System.ComponentModel.DataAnnotations;

namespace MLR.DAO;

public class LeakageAlert
{
    [Key]
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;

    public string RuleName { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

   
    public int Severity { get; set; } = 0;

    public decimal? Amount { get; set; }

    public bool IsRead { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public virtual ApplicationUser User { get; set; } = null!;
}
